xarray/tests/test_dataarray.py

```diff
@@ -4254,31 +4254,46 @@ def setup(self):
 
 
 @pytest.mark.parametrize(
-    "x, minindex, maxindex, nanindex",
+    ["x", "minindex", "maxindex", "nanindex"],
     [
-        (np.array([0, 1, 2, 0, -2, -4, 2]), 5, 2, None),
-        (np.array([0.0, 1.0, 2.0, 0.0, -2.0, -4.0, 2.0]), 5, 2, None),
-        (np.array([1.0, np.NaN, 2.0, np.NaN, -2.0, -4.0, 2.0]), 5, 2, 1),
-        (
+        pytest.param(np.array([0, 1, 2, 0, -2, -4, 2]), 5, 2, None, id="int"),
+        pytest.param(
+            np.array([0.0, 1.0, 2.0, 0.0, -2.0, -4.0, 2.0]), 5, 2, None, id="float"
+        ),
+        pytest.param(
+            np.array([1.0, np.NaN, 2.0, np.NaN, -2.0, -4.0, 2.0]), 5, 2, 1, id="nan"
+        ),
+        pytest.param(
             np.array([1.0, np.NaN, 2.0, np.NaN, -2.0, -4.0, 2.0]).astype("object"),
             5,
             2,
             1,
+            marks=pytest.mark.filterwarnings(
```

Comment:
I don't mind this but I would probably just have slapped a decorator on the function, supressing the warning in general.

```python
@pytest.mark.filterwarnings("ignore:invalid value encountered in reduce:RuntimeWarning")
```